aicsimageio/aics_image.py

```diff
@@ -1,242 +1,281 @@
-        self.dims = constants.DEFAULT_DIMENSION_ORDER
+        self.dims = Dimensions.DefaultOrder
 
-        # Lazy load later
+        # Determine reader class and create dask delayed array
+        reader_class = self.determine_reader(data=data)
+        self._reader = reader_class(data, **kwargs)
+
+        # Lazy load data from reader and reformat to standard dimensions
+        self._dask_data = None
         self._data = None
         self._metadata = None
 
-        # Determine reader class and load data
-        reader_class = self.determine_reader(data)
-        self._reader = reader_class(data, **kwargs)
+        # Store dask client and cluster setup
+        self._address = address
+        self._kwargs = kwargs
+        self._client = None
+        self._cluster = None
 
     @staticmethod
     def determine_reader(data: types.ImageLike) -> Type[Reader]:
-        """Cheaply check to see if a given file is a recognized type.
-        Currently recognized types are TIFF, OME TIFF, and CZI.
-        If the file is a TIFF, then the description (OME XML if it is OME TIFF) can be retrieved via read_description.
-        Similarly, if the file is a CZI, then the metadata XML can be retrieved via read_description.
-        """
-        # The order of the readers in this list is important.
-        # Example: if TiffReader was placed before OmeTiffReader, we would never use the OmeTiffReader.
-        for reader_class in [
-            NdArrayReader,
-            CziReader,
-            OmeTiffReader,
-            TiffReader,
-            DefaultReader,
-        ]:
+        """
+        Cheaply check to see if a given file is a recognized type and return the appropriate reader for the file.
+        """
+        # Iterate through the ordered supported readers to find the right one
+        for reader_class in SUPPORTED_READERS:
             if reader_class.is_this_type(data):
                 return reader_class
 
-        raise UnsupportedFileFormatError(type(data))
+        raise UnsupportedFileFormatError(data)
 
     @property
-    def data(self):
+    def dask_data(self) -> da.core.Array:
         """
         Returns
         -------
-        Returns a numpy.ndarray with dimension ordering "STCZYX"
+        Returns a dask array with dimension ordering "STCZYX"
         """
-        if self._data is None:
-            reader_data = self._reader.data
+        # Construct dask array if never before constructed
+        if self._dask_data is None:
+            reader_data = self.reader.dask_data
 
             # Read and reshape and handle delayed known dims reshape
-            self._data = transforms.reshape_data(
+            self._dask_data = transforms.reshape_data(
                 data=reader_data,
                 given_dims=self._known_dims or self.reader.dims,
                 return_dims=self.dims,
             )
+
+        return self._dask_data
+
+    @property
+    def data(self) -> np.ndarray:
```

Comment:
Docstrings here would be good. Apparently we never mention that you get the data from .data. 

we have imread methods but also .data (this isn't on you but) we should probably have some docs on when to use .data and when to use other functions.

Reply:
Does the README not handle this?